### PR TITLE
RD-5931 Docker image for NGINX

### PR DESCRIPTION
**Pull request by adrian-polanczyk-codilime**

### CODE REVIEW

**Review by tehasdf — COMMENTED** [review]
this whole `$DOLLAR` business is a bit ugly. It'll make it hard to work on the config files later.

But we don't have to do it!

1. I note that the nginx image has templating built-in, we don't need to do it ourselves, we just need to put templates `/etc/nginx/templates` (see [the docs](https://hub.docker.com/_/nginx), ctrl+f "templates")
2. So I suggest taking the smallest possible subset of things that need templating, putting them in templates, and including those templated files.

From what I see, this is only going to be the `worker_processes`, `worker_connections`, and `REST_SERVICE_PORT`? In which case, we can have just two templates:
- `workers.conf.template` containing `worker_processes` + `events { ...` - and an `include workers.conf;` in the main file
- `rest_upstream.conf.template` with just that `upstream`, included where it currently is

File: nginx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+WORKDIR /opt
+
+RUN rm -rf /etc/nginx/conf.d/default.conf
+
+RUN mkdir -p /var/log/cloudify/nginx
+
+COPY ./config/nginx.conf /etc/nginx/nginx.conf
+COPY ./config/conf.d /etc/nginx/conf.d
+COPY ./config/templates /etc/nginx/templates
```
Comment (mateumann) [inline nginx/Dockerfile:15]: Is `nginx/templates/` used anywhere?

File: nginx/config/conf.d/cloudify.conf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Declares the size of a request's body. This is mainly used to allow large blueprints to be uploaded.
+client_max_body_size 0m;
+
+# Overcoming the renowned http://stackoverflow.com/questions/19238122/nginx-with-ubuntu-and-rails-on-digital-ocean
+types_hash_max_size 4096;
+
+include "/etc/nginx/conf.d/*.upstream";
```
Comment (mateumann) [inline nginx/config/conf.d/cloudify.conf:7]: Hmm… I don't see any `*.upstream` files here.  Where are they?

File: nginx/config/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,37 @@
+user  nginx;
+include /etc/nginx/conf.d/workers.vars;
```
Comment (mateumann) [inline nginx/config/nginx.conf:2]: No such file or directory ;)

**Review by mateumann — APPROVED** [review]

**Review by mateumann — CHANGES_REQUESTED** [review]
please see the questions

**Review by mateumann — APPROVED** [review]

File: nginx/Dockerfile (hunk 1 above)
Comment (adrian-polanczyk-codilime) [reply nginx/Dockerfile:15]: files inside `templates` will be translated to ` conf.d` directory automatically

**Pull request merged**